rasa/core/channels/botframework.py

```diff
@@ -208,13 +208,23 @@ def blueprint(self, on_new_message):
                         postdata["recipient"],
                         postdata["serviceUrl"],
                     )
-
-                    user_msg = UserMessage(
-                        postdata["text"],
-                        out_channel,
-                        postdata["from"]["id"],
-                        input_channel=self.name(),
-                    )
+                    
+                    if postdata.get('attachments'):
+                        user_msg = UserMessage(
+                            text=(postdata['text'] if postdata.get('text') 
```

Comment:
```suggestion
                            text=postdata.get("text", "")
```

Reply:
I'm not sure I understand why this is necessary. what does it have to do with the presence of `postdata['attachments']`?

Reply:
if not, we could keep the creation of `user_msg` as it is, and add the attachment later if necessary:
```python
if postdata.get('attachments'):
    user_msg.metadata = {"attachments": postdata['attachments']}
```

Reply:
The thing is sometimes, something can be uploaded in the chat and Microsoft removes the `text` key of the json, making `postdata['text']` throw an error...